### ISSUE TITLE
508-defect-1 ❗ Launchblocker [SCREENREADER]: Text reminders checkbox alert MUST be announced to screen reader users and should be persistent

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#23168]
# [508-defect-1 :exclamation: Launchblocker](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-1)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Point of Contact

**VFS Point of Contact:** Josh

## User Story or Problem Statement

As a screen reader user, I expect any alerts that sighted users can see to be announced to me.

## Details

Upon unchecking the checkbox, a status update _temporarily_ replaces the content and isn't announced to screen reader users There's several issues with this when it comes to accessibility:
- Screen reader users will _not_ be able to access the status update because they won't even know it exists to begin with
- Even _if_ they are able to navigate to it, since it is temporary, it will disappear and will not be accessible again
- It's disorientating because it is removing and adding content from the page

## Acceptance Criteria

Example:
- [ ] The status update is persistent and does not disappear from the page
- [ ] The status update does not replace any content on the page
- [ ] The checkbox and related fields do not disappear
- [ ] Status updates are announced to screen reader users

## Steps to Recreate

My profile information, user 36 Wesley, upon unchecking the "we'll send VA reminders..." checkbox

## Proposed Solution (if known)

Attach `role="status"` to the status update so it can be announced to screen reader users. Make the status update persistent (it shouldn't disappear) and below the checkbox (it will be more discoverable).

## Screenshots or Trace Logs

https://user-images.githubusercontent.com/14154792/114613080-e3f9c800-9c70-11eb-9d0c-b0fb8e347b92.mov

**Comment by ajakabcin** [comment]
Designs for the proposed solution can be tracked in #24572

**Comment by ajakabcin** [comment]
@erikphansen final copy and designs for solution linked [here](https://github.com/department-of-veterans-affairs/va.gov-team/issues/24572#issuecomment-840642601)

**Comment by erikphansen** [comment]
@joshkimux We removed the checkbox and the new version should be on staging soon. Can you reevaluate how the mobile phone number works?

**Comment by joshkimux** [comment]
@erikphansen The launchblocking aspect of this ticket has been resolved 🥳 

That being said, two new issues have cropped up that I think should be relatively easy to address:
- Focus should be managed when the user activates "edit", focus should be sent to the first field so it is announced immediately and is visible to keyboard users with the yellow halo
- Instead of "dimmed", screen reader users should hear "updating your preferences" or something more descriptive to indicate a loading state

I've included a video that documents both!

https://user-images.githubusercontent.com/14154792/118843317-eff02f80-b897-11eb-8a49-ecf7a27c4a59.mov

**Comment by erikphansen** [comment]
Awesome, thanks for bringing up those two new (but also very old 😆 ) issues! I don't know why I never noticed the lack of putting focus in the first input as soon as you enter edit mode since that's a pet peeve of mine :(

@ajakabcin  can you one ticket for each of the issues that @joshkimux raised? They will actually apply to _all_ contact info fields, as well as the Direct Deposit info, I'd guess.

I think we can close this ticket, though.

**Comment by joshkimux** [comment]
Agreed we can close this out! 🥳

**Comment by ajakabcin** [comment]
Created #24952 and #24954 to address Josh's comments above. Thanks!

**Comment by sshein** [comment]
<3 everyone is so nice on this thread
